tests/utilities/test_functions.py

```diff
@@ -1,8 +1,19 @@
-import functools
+import pytest
 
 from compas.utilities import binomial
 
 
-def test_binomial():
-    assert binomial(3, 1) == 3
-    assert binomial(21, 10) == 352716
+@pytest.mark.parameterize(("n", "k", "result"),
+	[
+	pytest.param(3, 1, 3),
+	pytest.param(21, 10, 352716),
+	pytest.param(10, 0, 1),
+	pytest.param(0, 0, 1),
+	pytest.param(1, 2, 0, marks=pytest.mark.xfail(raises=ValueError)),
+	pytest.param(2, -1, 0, marks=pytest.mark.xfail(raises=ValueError)),
+	]
+)
+
```

Comment:
no spacing here, because the stuff above is a decorator for the function and should be "attached" to it...